src/main/java/bio/terra/service/job/JobMapKeys.java

```diff
@@ -1,29 +1,64 @@
 package bio.terra.service.job;
 
+import bio.terra.model.BillingProfileModel;
+import bio.terra.model.CloudPlatform;
+import bio.terra.service.iam.AuthenticatedUserRequest;
+import bio.terra.stairway.FlightFilter;
+import bio.terra.stairway.FlightFilterOp;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.exception.FlightFilterException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import java.util.UUID;
+import org.springframework.http.HttpStatus;
+
 public enum JobMapKeys {
   // parameters for all flight types
-  DESCRIPTION("description"),
-  REQUEST("request"),
-  REVERT_TO("revert_to"),
-  RESPONSE("response"),
-  STATUS_CODE("status_code"),
-  AUTH_USER_INFO("auth_user_info"),
-  SUBJECT_ID("subjectId"),
-  CLOUD_PLATFORM("cloudPlatform"),
+  DESCRIPTION("", new TypeReference<>() {}),
+  REQUEST,
+  REVERT_TO(new BillingProfileModel(), new TypeReference<>() {}),
+  RESPONSE,
+  STATUS_CODE(HttpStatus.OK, new TypeReference<>() {}),
+  AUTH_USER_INFO(new AuthenticatedUserRequest(), new TypeReference<>() {}),
+  SUBJECT_ID("", new TypeReference<>() {}),
+  CLOUD_PLATFORM(CloudPlatform.GCP, new TypeReference<>() {}),
```

Comment:
This part is confusing to me. Perhaps a small comment could be added? In particular, why are the parameter names now blank?

Reply:
The code now uses the enum names to generate the key names. Since the key names are now private, this seems OK to me; nothing outside the put/get is relying on these names.

Regarding the odd blank or empty parameters, this is how I enabled the type inference for `TypeReference<T>` to work. There has to be something in the constructor that references the type `T` that you want to capture.

While doing this it occurred to me that there's a few cases where, if a key is missing, a default value is used, and it seems like the code could use the argument as the default value in those cases. I didn't look at all uses so I'm not sure it will always work but it seems like a possible usage. Either way I think the code should gather up all the different types into another util class as constants that the code can use here.